hack/update-deps.sh

```diff
@@ -59,6 +59,16 @@ mkdir -p test/conformance/ingress
 rm -rf test/conformance/ingress/*
 rsync -r --no-perms "$SERVING_MODULE"/test/conformance/ingress/ test/conformance/ingress
 
+# Install yq command if it doesn't exist
+function install_yq() {
+  if [ -x "$(command -v yq)" ]; then
+    return
+  fi
+  go get github.com/mikefarah/yq/v3@3.3.0
```

Comment:
Should this be in hack/tools.go?

Reply:
We do not need any files from yq dependency so I dont think this should be part of hack/tools.go. WDYT?

Reply:
The point of `hack/tools.go` is to pull binary dependencies used in scripts like this into vendor (e.g. the K8s code-generators).  IMO we should follow a similar pattern of letting `go.mod` be the source of truth for version / pinning and changing this to just build the binary for use below.